cmd/tckgen.cpp

```diff
@@ -40,10 +40,11 @@
 using namespace MR;
 using namespace App;
 
-const char *algorithms[] = {
-    "fact", "ifod1", "ifod2", "nulldist1", "nulldist2", "sd_stream", "seedtest", "tensor_det", "tensor_prob", nullptr};
 #define DEFAULT_ALGORITHM 2 // ifod2
 
+const std::vector<std::string> algorithms = {
+    "fact", "ifod1", "ifod2", "nulldist1", "nulldist2", "sd_stream", "seedtest", "tensor_det", "tensor_prob"};
+
```

Comment:
It's quite common to have a list of strings define the set of options, but for both the default selection and the code branching based on user input to be based on integer index. While it's not requisite for this PR to go ahead, it's worth thinking about how this could perhaps be made less fragile.

Reply:
On this point, could we use enums? They seem to be most natural choice and we could provide functions to/from strings.

Reply:
I think I already use enums in some places for this kind of purpose; but it still nevertheless requires two parallel variables defined alongside one another, a strongly-typed enum and a list of strings. If there could be a way to both generate a list of strings from an enum for the CLI, and ideally to have the CLI class directly yield a corresponding enum instance when reading argument contents, that would be appealing, but it likely requires language capabilities I'm not familiar with.

Reply:
For enum to string conversion, we could use [magic_enum](https://github.com/Neargye/magic_enum), which is a popular header-only C++17 library.